torch/csrc/autograd/function.h

```diff
@@ -96,7 +98,11 @@ struct Function : std::enable_shared_from_this<Function> {
       edge_list&& next_edges = edge_list())
       : sequence_nr_(sequence_nr),
       num_inputs_(num_inputs),
-      next_edges_(std::move(next_edges)) {}
+      next_edges_(std::move(next_edges)) {
+    if (AnomalyMode::is_enabled()) {
+      AnomalyMode::store_stack(metadata());
```

Comment:
what does this do in a multithreaded environment?

Reply:
In a multithreaded environment, this saves the python traceback for the current thread.

From my understanding of how python works this is what I expect:
For python backed Functions, this will always be `.apply()` call for it.
For cpp backed Functions called from python, this will be the line containing the python operation that uses the C Function.
For cpp backed Functions called from C, it will be the python call that called into C if we're still on the same thread (for example backward of backward implemented in C will point to the first python backward call). If we changed thread (autograd on a different GPU), then you will get an empty traceback.